src/test/java/games/strategy/ui/SwingWorkerCompletionWaiterTest.java

```diff
@@ -1,24 +1,20 @@
 package games.strategy.ui;
 
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 
 import java.beans.PropertyChangeEvent;
 
 import javax.swing.SwingWorker;
 
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
 
-@RunWith(MockitoJUnitRunner.class)
 public final class SwingWorkerCompletionWaiterTest {
```

Comment:
So, I think the fact that it fails on Travis when you re-order the constructors suggests there is some non-determinism in how Mockito searches for a constructor to use for injection.  I re-ordered the constructors locally and was also able to trigger the exception.

I dug down into the Mockito internals until I got to the `ParameterizedConstructorInstantiator#biggestConstructor()` method:

```java
private Constructor<?> biggestConstructor(Class<?> clazz) {
    final List<? extends Constructor<?>> constructors = Arrays.asList(clazz.getDeclaredConstructors());
    Collections.sort(constructors, byParameterNumber);

    Constructor<?> constructor = constructors.get(0);
    checkParameterized(constructor, field);
    return constructor;
}
```

As you can see, it gets a list of all constructors, sorts them by parameter count, and then takes the first one.  Since `SwingWorkerCompletionWaiter` has two one-arg constructors, presumably the one that accepts a `Window` happens to get sorted first on one system and sorted second on another.  (Or, more likely, `Class#getDeclaredConstructors()` just returns the constructors in different orders on different systems, and the sort has no effect since both elements have the same parameter count.)  Basically, when the first constructor in the list is the `Window` constructor, it's as if the `ProgressWindow` constructor doesn't even exist (I verified that if I delete the `ProgressWindow` constructor altogether, I get the same exception).

So, in this case, I think avoiding `@InjectMocks` is justified.  This is a pretty subtle bug, so, if you don't mind, I'm going to submit a fix in a separate PR because I'd like to document this limitation of `@InjectMocks` in the commit message so we have it in our history.  I'll get it done before I go offline, and then you can merge it into this branch so you can change the runner to `StrictStubs` as part of this PR.

Thanks for bringing this problem to attention, @RoiEXLab.